modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -620,28 +840,173 @@ export default class ortcRTCPeerConnection extends yaeti.EventTarget {
+
+        // TODO: Should inspect the answer given by the app and update our
+        // sending RTP parameters if, for example, the app modified SSRC
+        // values.
+    }
+
+    /**
+     * Handles a local re-answer.
+     * @return {Promise}
+     * @private
+     */
+    _handleLocalReAnswer(desc) {
+        logger.debug('_handleLocalReAnswer(), desc:', desc);
+
+        const sdpObject = desc.sdpObject;
+
+        // Update local capabilities as decided by the app.
+        this._localCapabilities = sdpUtils.extractCapabilities(sdpObject);
+
+        logger.debug('local capabilities:', this._localCapabilities);
+
+        // TODO: Should inspect the answer given by the app and update our
+        // sending RTP parameters if, for example, the app modified SSRC
+        // values.
+    }
+
+    /**
+     * Handles the remote initial offer.
+     * @return {Promise}
+     * @private
+     */
+    _handleRemoteInitialOffer(desc) {
+        logger.debug('_handleRemoteInitialOffer(), desc:', desc);
+
+        const sdpObject = desc.sdpObject;
+
+        // Set MID values.
+        this._mids = sdpUtils.extractMids(sdpObject);
+
+        // Get remote RTP capabilities.
+        const remoteCapabilities = sdpUtils.extractCapabilities(sdpObject);
+
+        logger.debug('remote capabilities:', remoteCapabilities);
+
+        // Get local RTP capabilities (filter them with remote capabilities).
+        this._localCapabilities
+            = ortcUtils.getLocalCapabilities(remoteCapabilities);
+
+        // Start ICE and DTLS.
+        this._startIceAndDtls(desc);
+    }
+
+    /**
+     * Handles a remote re-offer.
+     * @return {Promise}
+     * @private
+     */
+    _handleRemoteReOffer(desc) {
+        logger.debug('_handleRemoteReOffer(), desc:', desc);
+
+        const sdpObject = desc.sdpObject;
+
+        // Update MID values (just in case).
+        this._mids = sdpUtils.extractMids(sdpObject);
+
+        // Get remote RTP capabilities (filter them with remote capabilities).
+        const remoteCapabilities = sdpUtils.extractCapabilities(sdpObject);
+
+        logger.debug('remote capabilities:', remoteCapabilities);
+
+        // Update local RTP capabilities (just in case).
+        this._localCapabilities
+            = ortcUtils.getLocalCapabilities(remoteCapabilities);
+    }
+
+    /**
+     * Promise based implementation for addIceCandidate().
+     * @return {Promise}
+     * @private
+     */
+    _addIceCandidate(candidate) { // eslint-disable-line no-unused-vars
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
         }
 
-        // TODO: More stuff to be done.
+        // NOTE: Edge does not support Trickle-ICE so just candidates in the
+        // remote SDP are applied.
+        logger.warn('addIceCandidate() not supported');
+
+        // Resolve the promise not to break things.
+        return Promise.resolve();
```

Comment:
if we are not supporting this, we should reject the promise. This function shouldn't be called anyway in non-P2P mode.